core/src/main/scala/org/http4s/MessageFailure.scala

```diff
@@ -83,8 +85,10 @@ final case class MalformedMessageBodyFailure(details: String, cause: Option[Thro
     s"Malformed message body: $details"
 
   def toHttpResponse[F[_]](httpVersion: HttpVersion): Response[F] =
-    Response(Status.BadRequest, httpVersion)
-      .withEntity(s"The request body was malformed.")(EntityEncoder.stringEncoder[F])
+    Message[Response].withEntity(
```

Comment:
Checking my own understanding: you're just avoiding syntax here, and the old version would have worked with implicits?